.github/workflows/deployment.yml

```diff
@@ -1,25 +1,321 @@
 name: Deployment
 
+concurrency: 
+  group: production
+  cancel-in-progress: true
+
 permissions:
-  contents: read
+  contents: write
 
 on:
   workflow_dispatch:
     inputs:
       tag_name:
         required: true
         type: string
+      environment:
+        default: production
+        type: environment
       go_version:
         default: "1.19"
         type: string
+      platforms:
+        default: "linux,macos,windows"
+        type: string
+      release:
+        description: "Whether to create a GitHub Release"
+        type: boolean
+        default: true
 
 jobs:
   linux:
     runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    if: contains(inputs.platforms, 'linux')
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ inputs.go_version }}
+      - name: Configure GoReleaser
+        run: |
+          touch CHANGELOG.md
+          sed '/#build:windows/,/^$/d; /#build:macos/,/^$/d' .goreleaser.yml >.goreleaser.generated.yml
```

Comment:
this makes me uncomfortable but agreed that keeping three different files in sync isn't great either.

Reply:
I've moved goreleaser config generation into a `script/release` script where the hack is more comprehensible.